helm/default-apps-aws/templates/apps.yaml

```diff
@@ -8,20 +8,32 @@ metadata:
     chart-operator.giantswarm.io/force-helm-upgrade: "{{ .forceUpgrade }}"
   labels:
     {{- include "labels.common" $ | nindent 4 }}
+    {{- if .inCluster }}  
+    app-operator.giantswarm.io/version: 0.0.0
+    {{- end }}
   name: {{ $.Values.clusterName }}-{{ $appName }}
   namespace: {{ $.Release.Namespace }}
 spec:
   catalog: {{ .catalog }}
   name: {{ .chartName }}
+  {{- if .inCluster }}  
+  namespace: {{ $.Release.Namespace }}
+  {{- else }}
   namespace: {{ .namespace }}
+  {{- end }}
   version: {{ .version }}
+  {{- if .inCluster }}
+  kubeConfig:
+    inCluster: true
+  {{- else }}
   kubeConfig:
     context:
       name: {{ $.Values.clusterName }}-admin@{{ $.Values.clusterName }}
     inCluster: false
     secret:
       name: {{ $.Values.clusterName }}-kubeconfig
       namespace: {{ $.Release.Namespace}}
+  {{- end }}
```

Comment:
Could we just use a single `if/else` here please? These templated are already a pain to debug.

Also, if making this change here can we please make sure the same functionality is replicated to `default-apps-gcp` also.

Reply:
We can probably do it yes :) let's first test that this works on CAPI and refactor

Reply:
You can test now before it's merged.

Reply:
can you check?